implementations/rust/ockam/ockam_command/src/util/mod.rs

```diff
@@ -496,15 +503,35 @@ pub fn setup_logging(verbose: u8, no_color: bool) {
             .with_default_directive(LevelFilter::TRACE.into())
             .parse_lossy(ockam_crates.map(|c| format!("{c}=trace")).join(",")),
     };
-    let fmt = fmt::Layer::default().with_ansi(!no_color);
-    let result = tracing_subscriber::registry()
+    let subscriber = tracing_subscriber::registry()
         .with(filter)
-        .with(tracing_error::ErrorLayer::default())
-        .with(fmt)
-        .try_init();
-    if result.is_err() {
-        eprintln!("Failed to initialise tracing logging.");
-    }
+        .with(tracing_error::ErrorLayer::default());
+    let (subscriber, guard) = match log_path {
+        None => {
+            let (n, guard) = tracing_appender::non_blocking(std::io::stdout());
+            let fmt = tracing_subscriber::fmt::Layer::default()
+                .with_ansi(!no_color)
+                .with_writer(n);
+            (subscriber.with(fmt).try_init(), guard)
+        }
+        Some(log_path) => {
+            let r = RollingFileAppender::new(
+                log_path,
+                RollingConditionBasic::new()
+                    .daily()
+                    .max_size(100 * 1024 * 1024), // Rotate every 100 MB or day
+                60, // Up to 6 GB or 60 days
+            )
+            .expect("Failed to create rolling file appender");
+            let (n, guard) = tracing_appender::non_blocking(r);
+            let fmt = tracing_subscriber::fmt::Layer::default()
+                .with_ansi(!no_color)
```

Comment:
Just for my own curiosity - when we spawn new subprocesses, are we setting no_color ourselves, or forwarding whatever setting was chosen in the original CLI command?

Reply:
Yes it does!